### PR TITLE
Phishlabs drp

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/19788

## Description
Partial implementation for DRP service of PhishLabs

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

**Comment by GalRabin** [comment]
@Arsenikr 
Hi,
Can you please approve the design so I will be able to start implementing it?

Thanks

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+                                                    date_format=occurred_format)
+    else:
+        datetime_new_last_run = last_run
+    # Query incidents by limit and creation time
+    total = 0
+    offset = 0
+    max_records = int(max_records)
+    max_records_page = 200
+    if max_records <= 200:
+        max_records_page = max_records
```
[the remark this answers is not in the document]
Comment (GalRabin) [reply Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: I did this mistake in the past, I will pay attention to it.

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 2]
```diff
@@ -307,9 +305,9 @@ def fetch_incidents_command(
                 'rawJSON': json.dumps(case_raw)
             })
 
-        new_last_run = cases_report[-1].get('occurred')
+        new_last_run = cases_report[0].get('occurred')
     # Return results
-    return cases_report, new_last_run
+    return cases_report, {"latsRun": new_last_run}
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: `lastRun`
unit tests should have caught that

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 3]
```diff
@@ -56,18 +63,73 @@ def get_cases(self, status: Optional[str] = None, case_type: Optional[str] = Non
...
+                                                params=assign_params(**params),
+                                                timeout=20)
+        cases_temp: List = raw_response.get('data', [])
+
+        format_time = "%Y-%m-%dT%H:%M:%SZ"
+        last_time: datetime = None if not cases_temp else datetime.strptime(cases_temp[0].get(date_field), format_time)
+        end_date_obj: datetime = datetime.strptime(end_date, format_time) if end_date else datetime.now()
+        begin_date_obj: Optional[datetime] = datetime.strptime(begin_date, format_time) if begin_date else None
+
+        index = 0
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: split the filtering code to another functions by logical units

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 4]
```diff
@@ -56,18 +63,73 @@ def get_cases(self, status: Optional[str] = None, case_type: Optional[str] = Non
...
+                    index = 0
+                    if not cases_temp:
+                        break
+                else:
+                    index += 1
+                last_time = datetime.strptime(cases_temp[index].get(date_field), format_time)
+            else:
+                break
+
+        # Move forward to end date but not exceed limit
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: what does that mean?

**Comment by GalRabin** [comment]
@kirbles19 
Can you please review my [yml](https://github.com/demisto/content/blob/a1264c5ad573e8b5049f08d3950de09562fac2ea/Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.yml)? thanks

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 5]
```diff
@@ -25,6 +29,9 @@
 INTEGRATION_COMMAND_NAME = 'phishlabs-ioc-drp'
 INTEGRATION_CONTEXT_NAME = 'PhishLabsIOC'
 
+# Disable insecure warnings
```
[the remark this answers is not in the document]
Comment (GalRabin) [reply Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: Checked, not implemented in client, so I need to leave it

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py (hunk 4 above)
Comment (GalRabin) [reply Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: Removed this line

File: Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py [hunk 6]
```diff
@@ -0,0 +1,562 @@
...
+            'type': case_type,
+            'offset': int(offset),
+            'maxRecords': int(max_records)
+        }
+        raw_response: Dict = self._http_request('GET',
+                                                url_suffix=suffix,
+                                                params=assign_params(**params),
+                                                timeout=20)
+        cases_temp: List = raw_response.get('data', [])
+        # About the drop some mean regex right now disable-secrets-detection-start
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_DRP/PhishLabsIOC_DRP.py]: there are no regexes and no secrets here, please remove the comment (same for the end comment)

**Comment by kirbles19** [comment]
@GalRabin Done. Lots of changes, please review carefully.

**Pull request merged**